### ISSUE TITLE
extract ReadableXXX to a standalone repo

### ISSUE THREAD

**Issue by BusyJay** [tikv/tikv#12755]
## Development Task

`ReadableXXX` structs are quite commonly used. Extracting them can be beneficial for all other repositories that wants to reuse the logic. For example, raft-engine can reuses the struct instead of reimplementing its own. We may also publish it to crates.io.

https://github.com/tikv/raft-engine/blob/812a8c0d0074837baa7f5340a942b614fa93437b/src/util.rs#L23-L49

**Comment by tonyxuqqi** [comment]
It's nice to have, but given raft-engine already paid the cost, it seems not a high priority work for now. I suggest we can revisit it when someone needs the same logic in other repo again.

**Comment by breezewish** [comment]
Not sure, but may be many things in the tikv_util could also be beneficial to the ecosystem :)

**Comment by BusyJay** [comment]
@breezewish Agree. ReadableXXX is just a start. And just publishing tikv_util to include all packages is not a good idea.
